src/types/privatekubectl.ts

```diff
@@ -75,13 +75,17 @@ export class PrivateKubectl extends Kubectl {
             runOutput
          )}`
       )
+
+      if (runOutput.exitCode !== 0) {
+         throw Error(
+            `Call to private cluster failed. Command: '${kubectlCmd}', errormessage: ${runOutput.stderr}`
+         )
+      }
+
       const runObj: {logs: string; exitCode: number} = JSON.parse(
          runOutput.stdout
       )
       if (!silent) core.info(runObj.logs)
-      if (runOutput.exitCode !== 0 && runObj.exitCode !== 0) {
```

Comment:
We need to keep this part here
```TypeScript
if (runOutput.exitcode !== 0) {
         throw Error(`failed private cluster Kubectl command: ${kubectlCmd}`)
}
```

Without this we won't behave exactly like the [overridden method](https://github.com/Azure/k8s-deploy/blob/e3e5e845b7c25d360b46bd6115c23e2f52a5ed9a/src/types/kubectl.ts#L173). `getExecOutput` will [throw if the exit code is none zero](https://github.com/actions/toolkit/blob/main/packages/exec/src/interfaces.ts#L28) so we need to replicate that behavior here.

Reply:
Looks good to me! I'll squash merge it when we merge into main. Thanks!